src/main/java/edu/agh/wfiis/designpatterns/patternsUsageExample/design_patterns/RefactorProvider.java

```diff
@@ -0,0 +1,5 @@
+package edu.agh.wfiis.designpatterns.patternsUsageExample.design_patterns;
+
+interface RefactorProvider {
```

Comment:
[Naming]
Maybe it's just me, but it sound weird to provide a verb.
It is a common practice in Java to name interfaces with an adjective. I think it is actually important because it lets you realize what shouldn't be an interface in the first place. Simply if it can't be named with adjective then it doesn't focus on providing functionality, so it shouldn't be an interface. For example you may want to create a Car interface, but it is impossible to name it with an adjective, because it describes too many functionalities. The adjective guideline prevents you from creating that interface and go for an abstract class that implements many interfaces each describing one car functionality, which is imo a better approach. 
So maybe CodeRefactoring or just Refactoring.